### ISSUE TITLE
Xcode project when building for simulators only sometimes runs tests

### ISSUE THREAD

**Issue by CoffeeFlux** [dotnet/runtime#50663]
I haven't the faintest idea what causes this.

When building, xharness fails to find the device, so I have to open the Xcode project and run it manually. However, pressing run in the project only actually runs it about half the time, and otherwise just builds and reports immediate success. There is no discernable difference between attempts, and rebuilding the test can fix or cause the issue seemingly at random.

cc: @steveisok

**Comment by steveisok** [comment]
I wondered about that too.  I had chalked it up as xcode "behavior".

**Comment by CoffeeFlux** [comment]
I ended up figuring out what's causing it at least. At the top of the Window there's a dropdown just to the left of the simulator type that appears to let you choose between a couple targets. If it's set to `ZERO_CEHCK` or `ALL_BUILD`, it won't run anything. I assume there's some way to set a default when generating the project, which we should do if so.

**Comment by EgorBo** [comment]
> I assume there's some way to set a default when generating the project, which we should do if so

Yeah, when I was adding xcode support I tried to find a way of changing the default target but with no luck

**Comment by steveisok** [comment]
I wonder if we could use one of the generated xcode projects as a template and tweak it to our liking?
